week06/populating_next_right_pointers_in_each_node.cpp

```diff
@@ -0,0 +1,16 @@
+// Time complexity: O(N), N = Number of nodes
+// Space complexity: O(1)
```

Comment:
Space complexity is O(H), where H = max depth of the tree. It is because of the stack memory used by the recursion.